### ISSUE TITLE
Large celery translate queue that is not clearing

### ISSUE THREAD

**Issue by RobinWeitzel** [WeblateOrg/weblate#8338]
### Describe the issue

My Weblate instance is very slow and after investigating I found that the DB was under constant load. I found that there are over 1000 translate tasks in the celery queue that seem to be stuck. The number is not reducing (even after waiting for days).
<img width="937" alt="image" src="https://user-images.githubusercontent.com/12135433/201464641-1e1db10c-fee7-426d-b8a9-776987d56268.png">
There are no errors in the logs and I update to the newest version to see if this helps (it did not). I also tried to clear the queue using 
 `docker exec weblate-docker weblate_ 1 celery -A weblate.utils purge -f`
However, celery stated that there are no messages to be purged:
![weblate_queue_3](https://user-images.githubusercontent.com/12135433/201464662-60cf5830-e0ed-4ca1-abd4-81ae942a9011.PNG)
Looking at the active and scheduled celery queues I also can not see over 1000 tasks:
![weblate_queue_2](https://user-images.githubusercontent.com/12135433/201464681-d2f6fa8b-5544-4f4a-a448-d2bf02485aee.PNG)
How can I get rid of the translate tasks?

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

Not clear what the root cause is, as such it is unclear how to reproduce this issue.

### Expected behavior

Translate queue shows 0 once translations are done.

### Screenshots

![weblate_queue](https://user-images.githubusercontent.com/12135433/201464713-44bf516e-118d-4155-9596-aff66ebdce97.PNG)
![weblate_queue_2](https://user-images.githubusercontent.com/12135433/201464714-e9d1db5f-870d-4d8d-85dc-e1ebb9795524.PNG)
![weblate_queue_3](https://user-images.githubusercontent.com/12135433/201464716-2114f43d-0ac6-4e7a-b881-dff0d378a0b9.PNG)

### Exception traceback

```pytb
No errors in the logs, no crashes.
```

### How do you run Weblate?

Docker container

**Comment by nijel** [comment]
There might be something wrong with the PostgreSQL database that made the queries hang.

The task queue cannot be cleared because the worker might have prefetched the queue. If you would stop the translate worker, the queue would be restored in redis and the management command would work.

For the next release, the performance should get slightly better due to d5a83bc217c0baf9917431e4f5629656bba680ea

**Comment by RobinWeitzel** [comment]
Hi @nijel, thanks for the reply :)

How would I stop the translate worker?
The queue keeps getting larger so I definitely want to find a solution.
I searched for it in the docs but could not find the right commands.
Will this have any other impacts if we do this?

**Comment by RobinWeitzel** [comment]
Hi @nijel ,

I tried the command but sadly this also did not do the trick.
Even after stopping the purge command still claimed that there were no messages in the queue.

However, when I restarted the celery task (with ```supervisorctl start celery-translate```) I noticed that the queue had decreased by 10. To confirm I stoped and started the task a second time and again the queue went down.

After further testing I found that repeatedly stoping and starting the task decreases the queue by 0-10 tasks each time.
Thus, I wrote a script that stoped and started the task 1000 times (waiting 10 seconds in between each command) and now the queue is empty :)

I will keep an eye on the queue over the next week and if I see it building up again I will set up a weekly script that stops and starts the task 10 times each day to purge anything stuck in the translate queue.

Thanks for your help!
